pkg/controller/container-runtime-config/helpers.go

```diff
@@ -459,10 +469,20 @@ func updatePolicyJSON(data []byte, internalBlocked, internalAllowed []string, re
 			transportScopes[reg] = signature.PolicyRequirements{
 				signature.NewPRReject(),
 			}
+			if r.FindStringIndex(reg) == nil {
+				deepNestedScopes[reg] = true
+			}
 		}
 	}
 
-	policyObj.Transports["atomic"] = transportScopes
+	// three or more path segments are not supported by the atomic transport
+	policyObj.Transports["atomic"] = make(signature.PolicyTransportScopes)
+	for reg := range transportScopes {
+		if deepNestedScopes[reg] {
```

Comment:
We don’t really need `deepNestedScopes`; move the check for a deeply-nested value check down here (and it will only need to be written once instead of twice).